class/defaults.yml

```diff
@@ -107,6 +112,7 @@ parameters:
         enabled: ${keycloak:ingress:enabled}
         annotations: ${keycloak:ingress:annotations}
         labels: ${keycloak:labels}
+        servicePort: https
```

Comment:
I don't think we should make this default. Similarly to encrypting db connection, we should make a how-to page and reference that page in the installation tutorial. We should definitely make enabling this as easy as possible though.
Besides, won't we have to tell the ingress controller to use https backend and ignore the self-signed cert that is installed on keycloak by default? (I checked the keycloak https backend: the cert is self-signed valid for 10 years)

e.g. via annotation like `nginx.ingress.kubernetes.io/backend-protocol: "HTTPS"`

Reply:
Sorry, but just ignoring a self-signed certificate is actually not what I expect from a security component.

I'm open to allow different configurations by default in general. I'm just be not sure, for who we should support it. I think it can't be of interest for us to deploy the component in an insecure way by default. From a solution standpoint I don't see any requirement having unencrypted support, but need to enforce a secure installation by default.

Reply:
It seems I fail to understand your goal. I thought you want to achieve something like this:

`HTTPS traffic -> Ingress terminates it with LE cert -> Ingress connects to https from Keycloak with self-signed cert`.

Did you mean to use the LE-provisioned certificates directly in Keycloak? Like

`HTTPS traffic -> Ingress (passthrough) -> Keycloak with LE-certs`

 In that case this PR is missing the changes or instructions how to mount those certificates from the secret or I'm missing something.

> I think it can't be of interest for us to deploy the component in an insecure way by default.

Your argument is valid, but then I'd expect more consistency and a how-to or tutorial how to install this component, since both acmecertificate and ingress are disabled by default atm. If we want to make the component secure by default, then we kind of have to make it opinionated, right?

Reply:
Ack. I will add my internal ticket notes in a proper way. Put this PR as long into draft mode.

Reply:
I've added the mappings to allow dynamically change the ingress variants. My proposed `passthrough` termination is still default, whenever other ingress configurations are still possible.

Reply:
`Reencrypt` is now default because the 3 month validity of the Let's encrypt certificate have a high potential to let outdate the loaded certificate, as Keycloak does not reload it (guess). The self singed certificate for the ingress backend has the same problem, but the duration of the certificate can be choose. So the problem is not solved, but mitigated in risk.

Another thing is that the certificate has to be issued manually, which is painful in a first place and the certificate is also not automatically renewed. We might replace that in a further version using https://cert-manager.io/docs/configuration/selfsigned/. For now it is good enough.